File: ussd.py
from flask import Flask, request

app = Flask(__name__)

@app.route('/ussd', methods=['POST'])
def ussd_callback():
    service_code = request.form.get('serviceCode')
    session_id = request.form.get('sessionId')
    user_phone = request.form.get('phoneNumber')
    text = request.form.get('text')

    if text == "":
        response = "CON what would you like to check \n"
        response += "1.  Check balance \n"
        response += "2. Make a payment \n"
    elif text == "1":
        response = "CON choose account information u want to view \n"
        response += "1. Account number \n"
        response += "2. Account balance \n"
    elif text == "2":
        response = "1. Please enter recipient's phone number \n"
    elif text == "1*1":
        response = "END your account number is: XXX-XXX"
    elif text == "1*2":
        response = "END your current balance is: yyyy"

    return response, 200, {'Content-Type': 'text/plain'}

if __name__ == '__main__':
    app.run(debug=True)

File: ussd.js
const express = require('express');
const bodyParser = require('body-parser');

const app = express();
const port = 3000;

app.use(bodyParser.urlencoded({ extended: false }));

app.post('/ussd', (req, res) => {
    const { serviceCode, sessionId, phoneNumber, text } = req.body;

    let response;

    if (text === "") {
        response = "CON what would you like to check \n";
        response += "1.  Check balance \n";
        response += "2. Make a payment \n";
    } else if (text === "1") {
        response = "CON choose account information u want to view \n";
        response += "1. Account number \n";
        response += "2. Account balance \n";
    } else if (text === "2") {
        response = "1. Please enter recipient's phone number \n";
    } else if (text === "1*1") {
        response = "END your account number is: XXX-XXX";
    } else if (text === "1*2") {
        response = "END your current balance is: yyyy";
    }

    res.status(200)
       .set('Content-Type', 'text/plain')
       .send(response);
});

app.listen(port, () => {
    console.log(`Server is running on port ${port}`);
});

